core/src/main/java/org/apache/accumulo/core/metadata/TabletFile.java

```diff
@@ -42,18 +43,31 @@
   private final TableId tableId; // 2a
   private final String tabletDir; // t-0003
   private final String fileName; // C0004.rf
-  private final String metadataEntry;
   private final Path metaPath;
   private final String normalizedPath;
 
+  private Optional<String> metadataEntry;
+
   /**
-   * Construct a tablet file using a Path object already created. Used in the case where we had to
-   * use Path object to qualify an absolute path.
+   * Construct new tablet file using a Path. Used in the case where we had to use Path object to
+   * qualify an absolute path or create a new file.
    */
-  public TabletFile(Path metaPath, String originalMetaEntry) {
-    this.metadataEntry = Objects.requireNonNull(originalMetaEntry);
+  public TabletFile(Path metaPath) {
```

Comment:
Recently, I've come to appreciate the use of static methods with a private constructor, rather than overloaded constructors.

For example:

```java
    var tabletFile = TabletFile.fromMetaPath(metaPath);
    var tabletFile = TabletFile.fromMetadataEntry(strEntry);
```

I recommend this over the overloaded constructors because you get to have very clear parameter semantics communicated via the method name, vs. trying to infer that from overloaded constructors. It's less error prone when parameter numbers mismatch, and easier to handle overloading when you have the same parameter type (usually String) with different semantics.

Reply:
Most of the time I would agree, especially with multiple String parameters a static method is a lot more clear.  But I don't think we need them in this case since the objects being passed are essentially the same thing just in different form.

Reply:
I don't see any downside, even if we don't "need" them, but I'll defer to your preference.